primitives/trie/src/storage_proof.rs

```diff
@@ -26,36 +29,74 @@ use sp_std::vec::Vec;
 /// The proof consists of the set of serialized nodes in the storage trie accessed when looking up
 /// the keys covered by the proof. Verifying the proof requires constructing the partial trie from
 /// the serialized nodes and performing the key lookups.
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[derive(Debug, PartialEq, Eq, Clone, Encode)]
 pub struct StorageProof {
-	trie_nodes: Vec<Vec<u8>>,
+	// TODO decode no more bytes to V0 for compatibility  and remove pub(crate)
+	pub(crate) trie_nodes: Vec<Vec<u8>>,
+	pub(crate) state_version: StateVersion,
```

Comment:
Changes proof format by adding state version.
Unresolve question, what is better:
- decode to V0 if state version not encoded, so previous code is guaranted unbroken.
- decode to V1 default if state version is not encoded, so it takes less size, but need coordinated update.
(for compatibility I can use a two byte header in the range of compact value encoding that does not exist by using  U8_OUT_OF_RANGE_VALUE for state version)